### ISSUE TITLE
Doesn't work if media.autoplay.blocking_policy is set to 2

### ISSUE THREAD

**Issue by mattjcleave** [Deskbot/No-Playlist-Autoplay-For-YouTube#1]

**Comment by Deskbot** [comment]
Thank you for the bug report. I believe you're referring to the flag in Firefox's "about:config".

What does setting "media.autoplay.blocking_policy" do? Can you point me to some documentation?

**Comment by mattjcleave** [comment]
> 
> 
> Thank you for the bug report. I believe you're referring to the flag in Firefox's "about:config".
> 
> What does setting "media.autoplay.blocking_policy" do? Can you point me to some documentation?

Sorry for the late reply. I forgot about it.

Yes, it's a flag in Firefox's about:config.

I found this:

This pref defines what the blocking policy would be used in blocking autoplay.
0 : use sticky activation (default)
https://html.spec.whatwg.org/multipage/interaction.html#sticky-activation
1 : use transient activation (the transient activation duration can be
    adjusted by the pref `dom.user_activation.transient.timeout`)
https://html.spec.whatwg.org/multipage/interaction.html#transient-activation
2 : user input depth (allow autoplay when the play is trigged by user input
    which is determined by the user input depth)

name: media.autoplay.blocking_policy
  type: uint32_t
  value: 0
  mirror: always

at https://searchfox.org/mozilla-central/rev/3ce874dc2703831af3e5ef3a1d216ffd08057fa5/modules/libpref/init/StaticPrefList.yaml#6353-6360

**Comment by Deskbot** [comment]
It seems to work for me. As in it autoplays when it should and doesn't autoplay when it shouldn't, although I have noticed other sites behaving weirdly.

What happens when you use it @himerche ?

**Comment by mattjcleave** [comment]
Sorry, I don't know why but it seems to be working now with the flag set to 2. It might not have worked once but I wasn't paying attention when it happened.

Before it consistently failed to stop autoplay in playlists when the flag was set to 2 but worked when I set it back to 0.

I just noticed that I have to restart Firefox for the changes to take effect after toggling autoplay so maybe that had something to do with it. But I might've tried restarting Firefox when it wasn't working correctly and I don't think I toggled autoplay before it stopped working.
